battleship-service/src/main/java/dreamteam/battleship/service/springcontroller/gamecontroller/GunSaluteController.java

```diff
@@ -0,0 +1,106 @@
+import dreamteam.battleship.service.springcontroller.model.Player;
+import dreamteam.battleship.service.springcontroller.model.response.ShootingResult;
+import dreamteam.battleship.service.springcontroller.model.response.TurnStatus;
+import dreamteam.battleship.service.springcontroller.util.PlayerQueue;
+import org.apache.log4j.Logger;
+
+import java.util.List;
+
+import static dreamteam.battleship.loggerhelper.LoggerStatics.END;
+import static dreamteam.battleship.loggerhelper.LoggerStatics.START;
+
+/**
+ * Created by daniel on 09.08.16.
+ */
+public class GunSaluteController extends GameControllerBase{
+
+    final static Logger logger = Logger.getLogger(GunSaluteController.class);
+
+
+    PlayerQueue playerQueue = new PlayerQueue();
+
+    public GunSaluteController(Player player1, MovementManager manager1) {
+        super(player1, manager1);
+    }
+
+    @Override
+    public boolean isMyTurn(Player player) {
+        return playerQueue.isMyTurn(player);
+    }
+
+    @Override
+    public ShootingResult handleShot(List<Integer> fieldNumbers, Player player) {
+
+        logger.debug(START);
+        ShootingResult response;
+        // check if there is sense to shoot
+        response = handleGunSaluteShoot(fieldNumbers, player);
+        logger.debug("shoot status is " + response.status);
+        logger.debug(END);
+
+        playerQueue.remove(player);
+        if(playerQueue.isEmpty())
+            playerQueue.add(player1, player2);
+
+        return response;
+    }
+
+    @Override
+    public void startGame() {
+        logger.debug(player1.name + " has board " + manager1.getBoard());
+        logger.debug(player2.name + " has board " + manager2.getBoard());
+        MovementManager tempManager1 = manager1;
+        MovementManager tempManager2 = manager2;
+        manager1 = new DamageManager(tempManager2.getBoard(), new MovementContainerImpl(), new ArbiterImpl(player2.shipList()));
+        manager2 = new DamageManager(tempManager1.getBoard(), new MovementContainerImpl(), new ArbiterImpl(player1.shipList()));
+        playerQueue.add(player1, player2);
+        isTheGameStarted = true;
+    }
+
+    @Override
+    public TurnStatus turnStatus(Player player) throws Exception {
+        MovementManager oppponentManager = getOpponentManager(player);
+        return new TurnStatus(getBoardForPlayer(player), isMyTurn(player), getWinner(), oppponentManager.numberOfPlayerShots());
+    }
+
+    private ShootingResult handleGunSaluteShoot(List<Integer> fieldNumbers, Player player) {
+        logger.debug("Handling the shoot");
+        ShootingResult response;
+
+        for(int fieldNumber: fieldNumbers) {
+            shotResponse(fieldNumber, player);
+        }
+
+        Player winner = getWinner();
+        MovementStatus ms  = winner != null ? MovementStatus.WON : MovementStatus.GUN_SALUTE_MODE;
+        response = new ShootingResult(ms, winner, getCurrentManager(player).getMovements());
+        return response;
```

Comment:
it easier to directly write 
return new ShootingResult(ms, winner, getCurrentManager(player).getMovements());